src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp

```diff
@@ -87,13 +87,20 @@ namespace
             return Model;
         }
 
-        virtual void prepare_inputs(void* data) const APPLESEED_OVERRIDE
+        virtual void prepare_inputs(
+            const ShadingPoint& shading_point,
+            void*               data) const APPLESEED_OVERRIDE
         {
             DipoleBSSRDFInputValues* values =
                 reinterpret_cast<DipoleBSSRDFInputValues*>(data);
 
             // Precompute the relative index of refraction.
-            values->m_eta = values->m_outside_ior / values->m_inside_ior;
+            const double outside_ior =
```

Comment:
I checked with a small sample scene and the debugger. It seems it does the right thing.